packages/builder/src/pages/builder/app/[application]/settings/oauth2/MoreMenuRenderer.svelte

```diff
@@ -1,11 +1,38 @@
 <script lang="ts">
-  import { ActionMenu, Icon, MenuItem } from "@budibase/bbui"
+  import { oauth2 } from "@/stores/builder"
+  import {
+    ActionMenu,
+    Icon,
+    MenuItem,
+    Modal,
+    notifications,
+  } from "@budibase/bbui"
+  import type { OAuth2Config } from "@budibase/types"
+  import OAuth2ConfigModalContent from "./OAuth2ConfigModalContent.svelte"
+  import { confirm } from "@/helpers"
+
+  export let row: OAuth2Config
+
+  let modal: Modal
 
   function onEdit() {
-    // TODO
+    modal.show()
   }
-  function onDelete() {
-    // TODO
+  async function onDelete() {
+    await confirm({
+      title: "Confirm Deletion",
+      body: `Deleting "${row.name}" cannot be undone. Are you sure?`,
+      okText: "Delete Configuration",
+      warning: true,
+      onConfirm: async () => {
+        try {
+          await oauth2.delete(row.id)
+          notifications.success(`Config '${row.name}' deleted successfully`)
+        } catch (e: any) {
+          notifications.error("Error deleting config")
```

Comment:
I think its always worth exposing the error in a scenario like this - its unlikely, but if it does happen it'll help diagnose what went wrong know what the message was.